### PR TITLE
Remote Keymanager: Proto Definitions and Basic Client Implementation

### PR DESCRIPTION
**Pull request by rauljordan**
**What type of PR is this?**

> Feature

**What does this PR do? Why is it needed?**

This PR begins defining the protobufs used by the remote keymanager v2 implementation in Prysm. This keymanager is a gRPC client that interacts with a remote server in order to fetch validating public keys and request signing of data.

**Which issues(s) does this PR fix?**

Part of #6220 

**DIFF LARGE DUE TO MOCKS**

### CODE REVIEW

File: proto/validator/accounts/v2/keymanager.proto [hunk 1]
```diff
@@ -11,4 +50,35 @@ message SignRequest {
 
     // Raw bytes data the client is requesting to sign.
     bytes data = 2;
```
Comment (terencechain) [inline proto/validator/accounts/v2/keymanager.proto:45]: I think we should rename `data` to `signing_root` to be consistent with the spec and avoid confusions.

In spec:
```
 bls.Sign(privkey, signing_root)
```

File: validator/accounts/v2/wallet.go [hunk 2]
```diff
@@ -10,9 +10,10 @@ import (
 
 	petname "github.com/dustinkirkland/golang-petname"
 	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
```
Comment (0xKiwi) [inline validator/accounts/v2/wallet.go:14]:
```suggestion
```

File: validator/keymanager/v2/remote/doc.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+relies on storing account information on-disk, making it trivial to import, export and
+list all associated accounts for a user.
+
+Package remote defines a keymanager implementation which connects to a remote signer
+server via gRPC. The connection is established via TLS using supplied paths to
+certificates and key files and allows for submitting remote signing requests for
+eth2 data structures as well as retrieving the available signing public keys from
+the remote server.
+
+Remote sign requests are defined by the following protobuf schema:
```
Comment (0xKiwi) [inline validator/keymanager/v2/remote/doc.go:15]: Does this need to be in the comment? Could be annoying to maintain.

File: validator/keymanager/v2/remote/remote_test.go [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+	"fmt"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+
```
Comment (0xKiwi) [inline validator/keymanager/v2/remote/remote_test.go:16]:
```suggestion
```

File: validator/keymanager/v2/remote/doc.go (hunk 3 above)
Comment (rauljordan) [reply validator/keymanager/v2/remote/doc.go:15]: It should be fine given how simple the proto message is. If it becomes more complex, agreed we should instead link to it. However, I do think it gives a nice picture to someone looking to write a a remote signer server as to what the protos should look like.

**Pull request merged**